graphql-ballerina/src/graphql/parser/lexer.bal

```diff
@@ -0,0 +1,325 @@
+            TokenType tokenType = getTokenType(char);
+            if (tokenType is SpecialCharacter) {
+                break;
+            } else if (tokenType is TerminalCharacter) {
+                break;
+            } else {
+                char = self.charReader.next();
+                check validateChar(char);
+                word += char.value;
+            }
+        }
+        TokenType 'type = getWordTokenType(word);
+        Scalar value = word;
+        if ('type is T_BOOLEAN) {
+            value = <boolean>'boolean:fromString(word);
+        }
+        return {
+            value: value,
+            'type: 'type,
+            location: location
+        };
+    }
+
+    isolated function pushToBuffer(Token token) {
+        self.buffer.push(token);
+    }
+}
+
+isolated function getTokenType(Char char) returns TokenType {
+    string value = char.value;
+    if (value is OPEN_BRACE) {
+        return T_OPEN_BRACE;
+    } else if (value is CLOSE_BRACE) {
+        return T_CLOSE_BRACE;
+    } else if (value is OPEN_PARENTHESES) {
+        return T_OPEN_PARENTHESES;
+    } else if (value is CLOSE_PARENTHESES) {
+        return T_CLOSE_PARENTHESES;
+    } else if (value is COLON) {
+        return T_COLON;
+    } else if (value is COMMA) {
+        return T_COMMA;
+    } else if (value is WhiteSpace) {
+        return T_WHITE_SPACE;
+    } else if (value is EOF) {
+        return T_EOF;
+    } else if (value is LineTerminator) {
+        return T_NEW_LINE;
+    } else if (value is QUOTE) {
+        return T_STRING;
+    } else if (value is Numeral) {
+        return T_INT;
+    } else if (value is HASH) {
+        return T_COMMENT;
+    }
+    return T_TEXT;
+}
+
+isolated function getToken(Scalar value, TokenType 'type, Location location) returns Token {
+    return {
+        'type: 'type,
+        value: value,
+        location: location
+    };
+}
+
+isolated function getTokenFromChar(Char charToken) returns Token {
+    TokenType tokenType = getTokenType(charToken);
+    return {
+        'type: tokenType,
+        value: charToken.value,
+        location: charToken.location
+    };
+}
+
+isolated function getWordTokenType(string value) returns TokenType {
+    if (value is Boolean) {
+        return T_BOOLEAN;
+    }
+    return T_TEXT;
+}
+
+isolated function getNumber(string value, boolean isFloat, Location location) returns int|float|InternalError {
+    if (isFloat) {
+        var number = 'float:fromString(value);
+        if (number is error) {
+            return getInternalError(value, "float", location);
+        } else {
+            return number;
+        }
+    } else {
+        var number = 'int:fromString(value);
+        if (number is error) {
+            return getInternalError(value, "int", location);
+        } else {
+            return number;
+        }
+    }
+}
+
+isolated function validateChar(Char char) returns InvalidTokenError? {
+    if (!stringutils:matches(char.value, VALID_CHAR_REGEX)) {
```

Comment:
Regex matching can be slow. If the perf is a concern, alternative is to call a native function, that does something like: https://github.com/ballerina-platform/ballerina-lang/blob/0eccf22fc7943c87d7017b8c8e723e5612db224d/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java#L1052

Reply:
Fixed with bb51408c4bf5de5c7fc6a0ef706d2f57626608ee